lib/wasm/WebAssemblyGenerator.js

```diff
@@ -237,6 +223,8 @@ const rewriteImportedGlobals = state => bin => {
 					t.instruction("get_local", [initialGlobalidx]),
 					t.instruction("set_global", [t.indexLiteral(newGlobals.length)])
 				);
+
+				additionalInitParams.push(t.funcParam(valtype, initialGlobalidx));
```

Comment:
I forgot to add the corresponding func args in https://github.com/webpack/webpack/pull/7413

Reply:
I don't think you forgot it. I think this PR isn't needed. `get_global` references an existing (imported) global, for which an param was already created.

Reply:
Interesting, I had trouble to find a test case for that, maybe you're right and it's not even needed.

The case is:
```wat
(module
  (import "a" "b" (global i32))
  (global i32 (get_global 0))
)
```

Without handling the global referencing another global, the init function will be:

```wat
(func (param i32)
  (get_local 0)
  (set_global 0)
)
```

And with is should look like this:

```wat
(func (param i32)
  (get_local 0)
  (set_global 0)
  (get_local 0)
  (set_global 1)
)
```

It seems to not be required. I need to recheck my tests.

Reply:
Just FYI an optimization here could be:

```wat
(func (param i32)
  (get_local 0)
  (tee_global 0) ;; tee_global keeps the value on the stack
  (set_global 1)
)
```

We just need to make sure to order the instructions correctly.